From da74befbcb02d4c86b07a6211d14e894a60d677e Mon Sep 17 00:00:00 2001
From: krunal soni <ksoni@codeaurora.org>
Date: Thu, 15 May 2014 14:35:00 -0700
Subject: [PATCH 2/2] wlan: HDD: Read operatingBand from cfg instead of ini

When iwpriv cmd is used to set the apOperatingBand and channel
range, it does not update the INI param apOperatingBand  but
updates cfg param WNI_CFG_SAP_CHANNEL_SELECT_OPERATING_BAND.
During start BSS the value of operatingBand is taken from INI.

Read the operatingBand value from cfg instead of
INI parameter during startBSS.

Crs-Fixed: 664886

Change-Id: Ia930a4b069c2026b932e1524b63498e40d18d438
---
 CORE/HDD/src/wlan_hdd_cfg80211.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/CORE/HDD/src/wlan_hdd_cfg80211.c b/CORE/HDD/src/wlan_hdd_cfg80211.c
index cfd75d5..85a05ea 100644
--- a/CORE/HDD/src/wlan_hdd_cfg80211.c
+++ b/CORE/HDD/src/wlan_hdd_cfg80211.c
@@ -2141,7 +2141,10 @@ static int wlan_hdd_cfg80211_start_bss(hdd_adapter_t *pHostapdAdapter,
          (sme_IsFeatureSupportedByDriver(DOT11AC)) &&
           (sme_IsFeatureSupportedByFW(DOT11AC)) )
     {
+        v_U32_t operatingBand = 0;
         pConfig->SapHw_mode = eSAP_DOT11_MODE_11ac;
+        ccmCfgGetInt(hHal, WNI_CFG_SAP_CHANNEL_SELECT_OPERATING_BAND,
+                     &operatingBand);
 
         /* If ACS disable and selected channel <= 14
          *  OR
@@ -2153,9 +2156,11 @@ static int wlan_hdd_cfg80211_start_bss(hdd_adapter_t *pHostapdAdapter,
          */
         if (((AUTO_CHANNEL_SELECT != pConfig->channel && pConfig->channel <= SIR_11B_CHANNEL_END)
                     || (AUTO_CHANNEL_SELECT == pConfig->channel &&
-                        iniConfig->apOperatingBand == RF_SUBBAND_2_4_GHZ)) &&
+                        operatingBand == RF_SUBBAND_2_4_GHZ)) &&
                 iniConfig->enableVhtFor24GHzBand == FALSE)
         {
+            hddLog(LOGW, FL("Setting hwmode to 11n, operatingBand = %d, Channel = %d"),
+                   operatingBand, pConfig->channel);
             pConfig->SapHw_mode = eSAP_DOT11_MODE_11n;
         }
     }
-- 
1.8.2.1

